examples/dwarfdump.rs

```diff
@@ -156,6 +156,7 @@ fn dump_entries<Endian>(mut entries: gimli::EntriesCursor<Endian>,
     let mut depth = 0;
     while let Some((delta_depth, entry)) = entries.next_dfs().expect("Should parse next dfs") {
         depth += delta_depth;
+        assert!(depth >= 0);
```

Comment:
Possibly it should print a nice error instead and clamp the depth to 0. You definitely don't want to let it go negative, even for release builds.